objcToxTests/OCTCallTimerTests.m

```diff
@@ -0,0 +1,79 @@
+//
+//  OCTCallTimerTests.m
+//  objcTox
+//
+//  Created by Chuong Vu on 6/25/15.
+//  Copyright (c) 2015 dvor. All rights reserved.
+//
+
+#import <UIKit/UIKit.h>
+#import <XCTest/XCTest.h>
+
+#import "OCTCallTimer.h"
+#import "OCTCall.h"
+#import "OCTRealmTests.h"
+
+@interface OCTCallTimer (Tests)
+
+@property (strong, nonatomic) dispatch_source_t timer;
+@property (weak, nonatomic) OCTRealmManager *realmManager;
+
+@end
+@interface OCTCallTimerTests : OCTRealmTests
+
+@property (strong, nonatomic) OCTCallTimer *callTimer;
+
+@end
+
+@implementation OCTCallTimerTests
+
+- (void)setUp
+{
+    [super setUp];
+
+    self.callTimer = [[OCTCallTimer alloc] initWithRealmManager:self.realmManager];
+    // Put setup code here. This method is called before the invocation of each test method in the class.
+}
+
+- (void)tearDown
+{
+    // Put teardown code here. This method is called after the invocation of each test method in the class.
+    [super tearDown];
+}
+
+- (void)testInit
+{
+    XCTAssertNil(self.callTimer.timer);
+    XCTAssertNotNil(self.callTimer.realmManager);
+}
+
+- (void)testStartTimer
+{
+    OCTFriend *friend = [self createFriend];
+    friend.friendNumber = 9;
+
+    [self.realmManager.realm beginWriteTransaction];
+    [self.realmManager.realm addObject:friend];
+    [self.realmManager.realm commitWriteTransaction];
+
+    OCTChat *chat = [self.realmManager getOrCreateChatWithFriend:friend];
+    OCTCall *call = [self.realmManager getOrCreateCallWithChat:chat];
+
+    //    XCTestExpectation *expectation = [self expectationWithDescription:@"Test Timer"];
+
+    [self.callTimer startTimerForCall:call];
+
+    NSInteger delayTime =  dispatch_time(DISPATCH_TIME_NOW, 1.5 * NSEC_PER_SEC);
+
+    //    dispatch_after(delayTime, dispatch_get_main_queue(), ^{
+    //        if (call.callDuration > 1) {
+    //            [expectation fulfill];
+    //        }
+    //    });
+
+    //    [self waitForExpectationsWithTimeout:3.0 handler:^(NSError *error) {
+    //        [self.callTimer stopTimer];
+    //    }];
```

Comment:
I don't think I implemented callTimer correctly since if I uncommented those lines, I will get:
`2015-06-25 22:02:36.392 xctest[12866:590373] *** Terminating app due to uncaught exception 'RLMException', reason: 'Realm accessed from incorrect thread'`

I have a feeling that my implementation of `- (void)startTimerForCall:(OCTCall *)call` is not putting it on the right thread? But shouldn't the below already take care of that?
`- (void)updateObject:(OCTObject *)object withBlock:(void (^)(id theObject))updateBlock`